config/cache/v1alpha1/cache.proto

```diff
@@ -63,6 +63,48 @@ message ResourceQuantity {
   string cpu = 2;
 }
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// Describes a data source connection. A map is available for passing implementation specific
+// properties.
+message DataSourceSpec {
+  // +kubebuilder:validation:Enum=POSTGRES_14;MYSQL_8;SQL_SERVER_2019
+  // Type and version of the underlaying DB. Needed to decide which drivers need to be used
+  optional DBType db_type = 1;
```

Comment:
If this field is missing, a default dbtype is used? In this case have you considered a zero value in the enum for `default`

Reply:
If the type is missing, the Cache  CR webhook prevents creation.

I don't think we  can have a default DB type as it's pure luck if that happens to be the user's DB type.

Reply:
I'm confused, if the field is `optional` then there is no default right? So either the field is present, in which case it has to be a valid value, or it is missing completely.

Reply:
sounds right, but the doc is confusing me [see](https://developers.google.com/protocol-buffers/docs/proto#optional)

Reply:
I think the rules are different for [proto3](https://developers.google.com/protocol-buffers/docs/proto3#specifying_field_rules):

```quote
optional: the same as singular, except that you can check to see if the value was explicitly set. An optional field is in one of two possible states:
the field is set, and contains a value that was explicitly set or parsed from the wire. It will be serialized to the wire.
the field is unset, and will return the default value. It will not be serialized to the wire.
```

So if the field is never set, nothing is serialized.

Reply:
It's still unclear to me the statement `the field is unset, and will return the default value`. We have some unit test on the java side that can check this, I would say we can merge and verify what happens exactly.